### PR TITLE
hide wiki edit button if user is not logged in , file _like.html.erb , Issue #11354

### LINKED ISSUE #11354: Hide Wiki Edit Button if User is not Logged In
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new** and looking through our general [help wanted](https://github.com/publiclab/plots2/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Claim the issue by commenting. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!
<details>
  <summary>💡 <i>Learn how to claim</i> 🙋</summary>
  <h3>Claiming an issue</h3>
  <p>Unless the issue is marked as reserved for someone, you can just say "I'd like to try this!" and then you've claimed it - no need to wait for someone to assign it to you. Just be sure you link your pull request (PR) to this issue so we can see where your solution is.</p>
  <p>And open one early if possible - even before you've completed it with additional commits - and others can help you figure out any issues you may face.</p>
</details>

- [ ] 📝 **Update** the file [app/views/like/_like.html.erb](https://github.com/publiclab/plots2/blob/main/app/views/like/_like.html.erb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -3,25 +3,25 @@
         <% if node.type == 'note' %>
           <span data-toggle="tooltip" data-placement="top" rel="tooltip" title="Edit Note">
             <a aria-label="Edit post" <% if current_user && current_user.uid != node.uid %>data-confirm='Please be aware that you are editing a post by another contributor, a privilege you have only because you are an admin or co-author.' <% end %> class='btn btn-outline-secondary btn-circle btn-sm' href='/notes/edit/<%= node.id %>?t=<%= Time.now.to_i %><% if params[:controller] == "questions" %>&redirect=question&template=question<% end %>'>
-                <i class='ml-1 ff fa fa-pencil'></i>
+                <i class='ff fa fa-pencil'></i>
             </a>
           </span>
         <% end %>
     <% end %>

-    <% if node.type == 'page' %>
-      <% if node.has_tag('locked') && (current_user ? current_user.basic_user? : true) %>
+    <% if node.type == 'page' && current_user %>
+      <% if node.has_tag('locked') && (current_user ? current_user.basic_user? : true) %>
         <a href="/wiki/locked" class='btn btn-outline-secondary btn-circle btn-sm' rel="tooltip" title="This page is locked. Click to learn why." data-placement="top">
-          <i class="ml-1 ff fa fa-lock"></i>
+          <i class="ff fa fa-lock"></i>
         </a>
-      <% elsif current_user && current_user.first_time_poster %>
+      <% elsif current_user.first_time_poster %>
         <a class='btn btn-outline-secondary btn-circle btn-sm' rel="tooltip" title="You can edit this wiki once your research note/question has been approved by moderators. Click to learn why." data-placement="top" href="https://publiclab.org/notes/tester/04-23-2016/new-moderation-system-for-first-time-posters">
-          <i class="ml-1 ff fa fa-lock"></i>
+          <i class="ff fa fa-lock"></i>
         </a>
       <% else %>
         <span data-toggle="tooltip" data-placement="top" rel="tooltip" title="Edit">
           <a class="btn-circle btn btn-outline-secondary" id="edit-btn" href='<%= node.edit_path %>?t=<%= Time.now.to_i %>'>
-          <i class='ml-1 ff fa fa-pencil'></i>
+            <i class='ff fa fa-pencil'></i>
           </a>
         </span>
       <% end %>
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there.

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming the `first-timers-only` issue. We're happy to keep it assigned to you as long as you need it if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance!

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below, and [reach out in our chatroom](https://publiclab.org/chat)!

### PR DESCRIPTION
**Pull request by nickmendezFlatiron**
Made changes specified in [Issue #11354](https://github.com/publiclab/plots2/issues/11354).
Removed the mentioned lines of code in _like.html.erb file and added the new lines of code.
Fixes #11354  

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [✔️ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by nickmendezFlatiron** [comment]
@publiclab/reviewers 
I made the changes mentioned but there is a tests/system-tests error. 

`=================[Screenshot]: tmp/screenshots/failures_test_edit_wiki.png
ERROR PostTest#test_edit_wiki (199.06s)
Minitest::UnexpectedError:         Capybara::ElementNotFound: Unable to find css "a.btn-circle:first-of-type .fa-pencil"
            test/system/post_test.rb:96:in block in <class:PostTest>'`

This error is referring to the test/system/post_test.rb file , the "edit wiki" test.

File: app/views/like/_like.html.erb [hunk 1]
```diff
@@ -3,25 +3,25 @@
...
+      <% elsif current_user.first_time_poster %>
         <a class='btn btn-outline-secondary btn-circle btn-sm' rel="tooltip" title="You can edit this wiki once your research note/question has been approved by moderators. Click to learn why." data-placement="top" href="https://publiclab.org/notes/tester/04-23-2016/new-moderation-system-for-first-time-posters">
-          <i class="ml-1 ff fa fa-lock"></i>
+          <i class="ff fa fa-lock"></i>
         </a>
       <% else %>
         <span data-toggle="tooltip" data-placement="top" rel="tooltip" title="Edit">
           <a class="btn-circle btn btn-outline-secondary" id="edit-btn" href='<%= node.edit_path %>?t=<%= Time.now.to_i %>'>
-          <i class='ml-1 ff fa fa-pencil'></i>
+          <i class="ff fa fa-lock"></i>
```
Comment (TildaDares) [inline app/views/like/_like.html.erb:24]: This should fix the failing test @nickmendezFlatiron. Sorry for the inconvenience!
```suggestion
          <i class="ff fa fa-pencil"></i>
```

**Review by TildaDares — CHANGES_REQUESTED** [review]
Feel free to just click the `Commit Suggestion` button. Thank you!

**Review by TildaDares — APPROVED** [review]
Amazing work @nickmendezFlatiron. Thank you!!

**Comment by TildaDares** [comment]
@jywarren Most of the tests are failing with this error
```
{'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Comment by TildaDares** [comment]
Congratulations on merging your PR @nickmendezFlatiron 🎉

**Comment by nickmendezFlatiron** [comment]
@TildaDares thank you for you help!

**Pull request merged**